### DISCUSSION TITLE
Is there a good way to use different versions of difficulty and performance calculation in a project?

### DISCUSSION THREAD

**Discussion by NafeeJ** [ppy/osu#23657] [Q&A] [answered]
I know if I want any given version I can install Osu.Game or whatever as a NuGet package but is there any clean way to use different versions? Do I just have to download the zip at whatever commit hash I want and delete everything except the files related to calculation? Are the different pushed live versions of calculations of different rulesets even documented anywhere?

Like if I wanted to do this in Javascript or something, others have made calculation ports such as rosu (live) or ojsama (old). I'm not really sure what would be the equivalent here.

**Comment by peppy** [comment]
Are you using the osu-tools project in this case? Can you describe your use case better? Referencing two different versions of the `osu.Game` assembly in a single project is probably going to be troublesome, but depending on what you're attempting to do there may be another way.

**Reply by NafeeJ** [reply]
i am attempting to write a program that reads the local data of stable (data/r, osu.db, scores.db) and displays the data. However, since its from stable, pp and star data isn't available without calculation (unless I'm mistaken?) which would be inaccurate anyway. What I wish to do is to be able to have both algorithms so I can see things like what my first several 3 star FCs were as I thought of them as 3 stars (since they may not be anymore) or what my top plays used to look like at any point in time. 

i was looking at osu-tools for reference on how to actually use the code though.

**Reply by peppy** [reply, marked as answer]
I think your best bet, assuming you're comfortable with getting things hooked up, would be to use `AssemblyLoadContext`, which will (in theory) let you dynamically load an unload assemblies into your app.

Using this you could create a folder with the different `osu.Game.dll` versions you want to calculate against and iterate through them.

**Reply by NafeeJ** [reply]
yeah that sounds good thanks
